### PR TITLE
[Keras Sprint] Add KerasCV visualizations and augmentations to 'Image Classification From Scratch' example

### PR DESCRIPTION
**Pull request by suvadityamuk**
This PR fixes the task as part of the project raised during Keras Community Sprint. Some important notes to add:

- It adds a KerasCV dependency to the example
- Makes a few changes to the text of the tutorial
- Uses `keras_cv.visualizations.plot_image_gallery` to visualize images, instead of the original manual implementation
- Uses `keras_cv.layers.RandomFlip` and `keras_cv.layers.RandAugment` in sequential format, for data augmentation.

Please let me know if any additions or changes should be made, to align it with KerasCV more.
@martin-gorner @ianstenbit @fchollet

### CODE REVIEW

File: examples/vision/image_classification_from_scratch.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Title: Image classification from scratch
 Author: [fchollet](https://twitter.com/fchollet)
```
Comment (martin-gorner) [inline examples/vision/image_classification_from_scratch.py:3]: Change author to:
"[fchollet](https://twitter.com/fchollet), updated by <names of the updaters>"

File: examples/vision/image_classification_from_scratch.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 Title: Image classification from scratch
```
Comment (martin-gorner) [inline examples/vision/image_classification_from_scratch.py:2]: Change title to:
"[KerasCV] Image classification from scratch"

File: examples/vision/image_classification_from_scratch.py [hunk 3]
```diff
@@ -99,19 +100,33 @@
 """
 ## Visualize the data
 
-Here are the first 9 images in the training dataset. As you can see, label 1 is "dog"
-and label 0 is "cat".
+Here are the first 8 images in the training dataset. We visualize this by making
```
Comment (martin-gorner) [inline examples/vision/image_classification_from_scratch.py:103]: I suggest to shorten this description:
"Here are the first 8 images in the training dataset, visualized using the KerasCV `plot_image_gallery` utility."

**Review by martin-gorner — CHANGES_REQUESTED** [review]
cosmetic changes, good to go otherwise

File: examples/vision/image_classification_from_scratch.py (hunk 1 above)
Comment (suvadityamuk) [reply examples/vision/image_classification_from_scratch.py]: Done!

File: examples/vision/image_classification_from_scratch.py (hunk 2 above)
Comment (suvadityamuk) [reply examples/vision/image_classification_from_scratch.py]: Done!

**Review by martin-gorner — APPROVED** [review]
LGTM

**Review by martin-gorner — APPROVED** [review]

**Review by ianstenbit — APPROVED** [review]
This lgtm -- one thing to consider is including a link to https://keras.io/guides/keras_cv/classification_with_keras_cv if we're going to use KerasCV here

**Comment by suvadityamuk** [comment]
> This lgtm -- one thing to consider is including a link to https://keras.io/guides/keras_cv/classification_with_keras_cv if we're going to use KerasCV here

Where do you think I should link this?

I'll add the generated files in a while too!

**Comment by ianstenbit** [comment]
> > This lgtm -- one thing to consider is including a link to https://keras.io/guides/keras_cv/classification_with_keras_cv if we're going to use KerasCV here
> 
> Where do you think I should link this?
> 
> I'll add the generated files in a while too!

I'll let @fchollet weigh in on that as he owns this example. It may be that it's not worth including, but I figured something at the end that says "For more information about training image classifiers in KerasCV, see []..."

**Comment by suvadityamuk** [comment]
Sure, sounds good to me. In that case, I'll wait for @fchollet to weigh in on this and just verify the entire thing, before adding the generated files

**Comment by suvadityamuk** [comment]
Hi, @fchollet!

Gentle ping for a quick review!

**Comment by suvadityamuk** [comment]
Hi, @fchollet!

Let me know if this looks good! I'll add the generated files in that case

**Review by fchollet — APPROVED** [review]
LGTM, thank you for the contribution!

**Pull request merged**